### PR TITLE
test: Add tests for uncovered API functions.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/api/java/io/github/cvc5/Proof.java [hunk 1]
```diff
@@ -20,39 +20,28 @@
 /**
  * A cvc5 Proof.
  */
-public class Proof implements IPointer
+public class Proof extends AbstractPointer implements Comparable<Proof>
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/Proof.java:23]: IPointer is necessary for avoiding memory leak and Comparable<Proof> is not needed here. To pass coverage tests, the method equals should be explicitly called.

File: src/api/java/io/github/cvc5/Proof.java [hunk 2]
```diff
@@ -20,39 +20,28 @@
...
   {
-    this(getNullProof());
+    super(getNullProof());
   }
 
   private static native long getNullProof();
 
   Proof(long pointer)
   {
-    this.pointer = pointer;
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/Proof.java:37]: Keep this code

File: src/api/java/io/github/cvc5/Proof.java [hunk 3]
```diff
@@ -20,39 +20,28 @@
...
-    {
-      deletePointer(pointer);
-    }
-    pointer = 0;
+    super(pointer);
   }
 
   protected native void deletePointer(long pointer);
 
-  public long getPointer()
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/Proof.java:53]: ditto

File: src/api/java/io/github/cvc5/Proof.java [hunk 4]
```diff
@@ -138,6 +127,21 @@ public boolean equals(Object p)
...
+  /**
+   * Comparison for ordering on proofs.
+   *
+   * @param p The proof to compare to.
+   * @return A negative integer, zero, or a positive integer if this proof
+   *         is referentially less than, equal to, or greater than the
+   *         specified proof.
+   */
+  @Override
+  public int compareTo(Proof p)
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/Proof.java:139]: Not needed

File: src/api/java/io/github/cvc5/Proof.java (hunk 1 above)
Comment (aniemetz) [reply src/api/java/io/github/cvc5/Proof.java:23]: AbstractPointer implements IPointer. This is also what we do for sorts and terms. Operator `==` should call `equals`, else we get an issue in the future if we change the comparison from pointer comparison to something more sophisticated (not that I'm saying that we plan to do this in the near future).

File: test/unit/api/java/ProofTest.java [hunk 5]
```diff
@@ -103,6 +117,25 @@ void nullProof() throws CVC5ApiException
...
   }
 
+  @Test
+  void equalHash() throws CVC5ApiException
+  {
+    Proof x = createProof();
+    Proof y = x.getChildren()[0];
+    // Proof z = createProof2();
+    Proof n = new Proof();
+    assertTrue(x == x);
```
Comment (mudathirmahgoub) [inline test/unit/api/java/ProofTest.java:127]:
```suggestion
    assertTrue(x.equals(x));
```

File: src/api/java/io/github/cvc5/Proof.java (hunk 2 above)
Comment (aniemetz) [reply src/api/java/io/github/cvc5/Proof.java:37]: This should be covered by the super call.

File: src/api/java/io/github/cvc5/Proof.java (hunk 3 above)
Comment (aniemetz) [reply src/api/java/io/github/cvc5/Proof.java:53]: ditto

File: src/api/java/io/github/cvc5/Proof.java (hunk 4 above)
Comment (aniemetz) [reply src/api/java/io/github/cvc5/Proof.java:139]: It is when implementing comparable.

File: test/unit/api/java/ProofTest.java (hunk 5 above)
Comment (aniemetz) [reply test/unit/api/java/ProofTest.java:127]: See my comment above.

File: src/api/java/io/github/cvc5/Proof.java (hunk 1 above)
Comment (aniemetz) [reply src/api/java/io/github/cvc5/Proof.java]: Resolved this offline.

File: src/api/java/io/github/cvc5/Grammar.java [hunk 6]
```diff
@@ -41,6 +41,17 @@ public Grammar(Grammar grammar)
 
   // endregion
 
+  /**
+   * Determine if this is the null grammar.
+   * @return True if this Sort is the null grammar.
```
Comment (mudathirmahgoub) [inline src/api/java/io/github/cvc5/Grammar.java:46]:
```suggestion
   * @return True if this Grammar is the null grammar.
```

**Review by mudathirmahgoub — APPROVED** [review]
LGTM

**Pull request merged**